src/test/java/com/google/cloud/spanner/pgadapter/StatementTest.java

```diff
@@ -300,4 +300,31 @@ public void testPortalStatementDescribePropagatesFailure() throws Exception {
 
     intermediateStatement.describe();
   }
+
+  @Test
+  public void testBatchStatements() throws Exception {
+    String sql =
+        "INSERT INTO users (id) VALUES (1); INSERT INTO users (id) VALUES (2);INSERT INTO users (id) VALUES (3);";
```

Comment:
Also add a test for something like:

1. Double ';', e.g: SELECT 1;; SELECT 2;
Would this work? with the current code we might send an empty query, would the backend work with this?
2. ';' within comments
3. Perhaps at least one e2e test with real spanner

Reply:
I added some additional test for the other edge cases.

I have an e2e DML test and an e2e DDL test.